infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -372,6 +378,79 @@ public synchronized void resetServer(CorfuPayloadMsg<SequencerTailsRecoveryMsg>
         r.sendResponse(ctx, msg, CorfuMsgType.ACK.msg());
     }
 
+    @ServerHandler(type = CorfuMsgType.TOKEN_BATCH_REQ)
+    public synchronized void batchRequest(CorfuPayloadMsg<BatchTokenRequest> msg,
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 22 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)